test/DynamicSale.test.ts

```diff
@@ -99,20 +99,19 @@ describe('DynamicSale', () => {
   });
 
   describe('calculateTotalCost', () => {
-    it.skip('works', async () => {
+    it.only('works', async () => {
       const context = await setup();
 
       const user = setSigner(context.users[0], context);
 
       await context.deployer.TDFToken.mint(context.deployer.address, parseEther('2433'));
 
-      // await user.calculateTotalCost('1').toEq('223.12');
-      // await user.calculateTotalCost('5').toEq('1126.77');
-      // await user.calculateTotalCost('10').toEq('2281.98');
-      // await user.calculateTotalCost('20').toEq('4680.66');
-      // await user.calculateTotalCost('30').toEq('7202.0');
-      await user.calculateTotalCost('100').toEq('28855.65');
-      // await user.calculateTotalCost('5000').toEq('28855.641455224455');
+      await Promise.all(
```

Comment:
Promise.all expects to receive a list of `Promise` s you are returning the data. does this work?

Reply:
in general this is what we should test

Reply:
hey @arturictus, thanks for coming to the pr! appreciate it. 

Yes, `Promise.all` is how you deal with async functions in the `.map`. To be honest it would be the same as `dynamicPriceMock.forEach`

No, it does not work, and I don't know why:

```
Error: missing revert data in call exception
```

Reply:
you can see the full error log in this PR checks